### PR TITLE
[!!!][TASK] Add PHP 8 compatibility, raise minimum PHP version to 7.3

### PR DESCRIPTION
**Pull request by patrickhilker**
Hey,

I would like to use this lib in a PHP8 project, so I added support for it. Unfortunately  I had to drop support for PHP < 7.3 to upgrade the dependencies as needed.

I would love to see this merged. As it is breaking, I assume there has to be a new major release.

Please provide any feedback you have!

Patrick

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -1,24 +1,13 @@
...
-  convertErrorsToExceptions="true"
-  convertWarningsToExceptions="true"
-  stopOnError="false"
-  stopOnFailure="false"
-  stopOnIncomplete="false"
-  stopOnSkipped="false"
-  verbose="false">
-
+<?xml version="1.0"?>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" bootstrap="vendor/autoload.php" colors="true" convertErrorsToExceptions="true" convertWarningsToExceptions="true" stopOnError="false" stopOnFailure="false" stopOnIncomplete="false" stopOnSkipped="false" verbose="false" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd">
```
Comment (mbrodala) [inline phpunit.xml:2]: Can you restore the previous formatting? This makes it easier to keep track of future changes.

File: phpunit.xml [hunk 2]
```diff
@@ -1,24 +1,13 @@
...
+  <coverage>
```
Comment (mbrodala) [inline phpunit.xml:3]: Can you move this after the `testsuites`?

**Review by mbrodala — COMMENTED** [review]
Thanks a lot for tackling this! I'm glad that there are actually people using this library outside of ourselves. :-)

A small request: please amend the commits to adhere to the [TYPO3 commit message rules](https://docs.typo3.org/m/typo3/guide-contributionworkflow/master/en-us/Appendix/CommitMessage.html).

**Comment by patrickhilker** [comment]
Found some time today to fix your review comments and squashed my commits. I hope this fits to the rules.

**Comment by mbrodala** [comment]
Just did a rebase to wake up Github Actions. Looks great now, thanks for the patience!

**Review by mbrodala — APPROVED** [review]

**Comment by patrickhilker** [comment]
Thanks for merging!

Could you please create a new release? Or do you have more in mind for a new release?

**Comment by mbrodala** [comment]
I only had to bump our internal project accordingly.

Version 2.1.0 with your change is out now.

**Pull request merged**